blueprints/ember-cli-blueprint-test-helpers/index.js

```diff
@@ -1,12 +1,97 @@
-    ]);
+      this.insertTestCallToPackage(options)
+    ];
+    var packages = [{name: 'mocha', target: '^2.2.1'}];
+
+    if (options.babel) {
+      afterInstallTasks.splice(0, 0,
+        this.insertIntoJsonFile('./node-tests/.babelrc',{plugins: ["transform-es2015-arrow-functions", "transform-es2015-shorthand-properties"]})
+      );
+      // add babel specific packages
+      packages.splice(0, 0, {name: 'babel-plugin-transform-es2015-arrow-functions', target: '^6.5.2'},
+          {name: 'babel-plugin-transform-es2015-shorthand-properties', target: '^6.5.0'},
+          {name: 'babel-register', target: '^6.7.2'});
+    }
+    afterInstallTasks.splice(0, 0, this.addPackagesToProject(packages));
+    return Promise.all(afterInstallTasks);
+  },
+  insertIntoJsonFile: function(pathRelativeToProjectRoot, contents) {
+    var fullPath = path.join(this.project.root, pathRelativeToProjectRoot);
+    var contentsToWrite;
+    var originalContents = {};
+    if (existsSync(fullPath)) {
+      originalContents = JSON.parse(fs.readFileSync(fullPath, { encoding: 'utf8' }));
+    }
+    contentsToWrite = JSON.stringify(merge(originalContents, contents), null, 2);
+    var returnValue = {
+      path: fullPath,
+      originalContents: originalContents,
+      contents: contentsToWrite,
+      inserted: false
+    };
+    if (contentsToWrite !== originalContents) {
+      returnValue.inserted = true;
+
+      return writeFile(fullPath, contentsToWrite)
+        .then(function() {
+          return returnValue;
+        });
+    } else {
+      return Promise.resolve(returnValue);
+    }
+  },
+  insertTestCallToPackage: function(options) {
+    var contentsToWrite;
+    var fullPath = path.join(this.project.root, 'package.json');
+    var scriptContents = 'mocha node-tests --recursive';
+    var babelContents = ' --require babel-register';
+    var originalContents = fs.readFileSync(fullPath, {encoding: 'utf8' });
+    var updatedPackage = JSON.parse(originalContents);
+    if (updatedPackage.scripts.nodetest === scriptContents) {
+      if (options.babel) {
+        updatedPackage.scripts.nodetest += babelContents;
+      }
+    } else if (typeof updatedPackage.scripts.nodetest === 'undefined') {
+      updatedPackage.scripts.nodetest = scriptContents + (options.babel ? babelContents : '');
+    } else {
+      this.ui.writeLine('Could not update "nodetest" script in package.json. Please add "' + scriptContents +
+        (options.babel ? babelContents : '') + '" to your nodetest script.');
+    }
+    contentsToWrite = JSON.stringify(updatedPackage, null, 2);
+    var returnValue = {
+      path: fullPath,
+      originalContents: originalContents,
+      contents: contentsToWrite,
+      inserted: false
+    };
+    if (contentsToWrite !== originalContents) {
+      returnValue.inserted = true;
+
+      return writeFile(fullPath, contentsToWrite)
+        .then(function() {
+          return returnValue;
+        });
```

Comment:
Seems that some sort of race condition happens if I change it to all sync, and there's no reason to read async, so I'm going to leave it as is.